tests/components/samsungtv/test_media_player.py

```diff
@@ -55,250 +117,405 @@ class UnhandledResponse(Exception):
-        self.device.volume_up()
-        self.device.send_key.assert_called_once_with("KEY_VOLUP")
-
-    def test_volume_down(self):
-        """Test for volume_down."""
-        self.device.send_key = mock.Mock()
-        self.device.volume_down()
-        self.device.send_key.assert_called_once_with("KEY_VOLDOWN")
-
-    def test_mute_volume(self):
-        """Test for mute_volume."""
-        self.device.send_key = mock.Mock()
-        self.device.mute_volume(True)
-        self.device.send_key.assert_called_once_with("KEY_MUTE")
-
-    def test_media_play_pause(self):
-        """Test for media_next_track."""
-        self.device.send_key = mock.Mock()
-        self.device._playing = False
-        self.device.media_play_pause()
-        self.device.send_key.assert_called_once_with("KEY_PLAY")
-        assert self.device._playing
-        self.device.send_key = mock.Mock()
-        self.device.media_play_pause()
-        self.device.send_key.assert_called_once_with("KEY_PAUSE")
-        assert not self.device._playing
-
-    def test_media_play(self):
-        """Test for media_play."""
-        self.device.send_key = mock.Mock()
-        self.device._playing = False
-        self.device.media_play()
-        self.device.send_key.assert_called_once_with("KEY_PLAY")
-        assert self.device._playing
-
-    def test_media_pause(self):
-        """Test for media_pause."""
-        self.device.send_key = mock.Mock()
-        self.device._playing = True
-        self.device.media_pause()
-        self.device.send_key.assert_called_once_with("KEY_PAUSE")
-        assert not self.device._playing
-
-    def test_media_next_track(self):
-        """Test for media_next_track."""
-        self.device.send_key = mock.Mock()
-        self.device.media_next_track()
-        self.device.send_key.assert_called_once_with("KEY_FF")
-
-    def test_media_previous_track(self):
-        """Test for media_previous_track."""
-        self.device.send_key = mock.Mock()
-        self.device.media_previous_track()
-        self.device.send_key.assert_called_once_with("KEY_REWIND")
-
-    def test_turn_on(self):
-        """Test turn on."""
-        self.device.send_key = mock.Mock()
-        self.device._mac = None
-        self.device.turn_on()
-        self.device.send_key.assert_called_once_with("KEY_POWERON")
-        self.device._wol.send_magic_packet = mock.Mock()
-        self.device._mac = "fake"
-        self.device.turn_on()
-        self.device._wol.send_magic_packet.assert_called_once_with("fake")
+@pytest.fixture(name="remote")
+def remote_fixture():
+    """Patch the samsungctl Remote."""
+    with patch("samsungctl.Remote") as remote_class, patch(
+        "samsungctl.exceptions"
+    ) as exceptions_class, patch(
+        "homeassistant.components.samsungtv.media_player.socket"
+    ) as socket_class:
+        remote = mock.Mock()
+        remote_class.return_value = remote
+        exceptions_class.AccessDenied = AccessDenied
```

Comment:
We should use the real exceptions instead. I think we can remove this patch. It was needed before when we didn't have the library available.